### DISCUSSION TITLE
Material properties FEM

### DISCUSSION THREAD

**Discussion by JeffPeter** [idaholab/moose#16262] [Q&A General] [answered]
Hi,

I'm a new user of MOOSE and using FEM in general. Can you explain me the difference between nodes and elements? If I understood the documentation correctly, the equations are solved at the nodes and interpolated in the elements?  

Are the material properties also assigned at the nodes? 

Thanks so much for the nice documentation and help
Jeff

**Comment by crswong888** [comment, marked as answer]
Correct, equations are solved at the nodes and numerically integrated/polynomial interpolated through the elements (between nodes). Material properties are assigned to the elements and are coefficients in the element integrals. It is possible to also define material properties that vary throughout an element space, but still material properties are defined on elements, not nodes.

Consider checking out this page: [https://mooseframework.inl.gov/help/faq/what_is_fem.html](https://mooseframework.inl.gov/help/faq/what_is_fem.html)

**Reply by dschwen** [reply]
I think it is misleading to say that "equations are solved at the nodes". We happen to use nodal basis functions, which have their DOFs at the nodes. But we are solving for a projection of the true solution onto a lower dimensional function space with piecewise continuous functions. There also isn't really any "interpolation", instead the solution in this lower dimensional function space is defined _at every point_. We're just evaluating it.

**Reply by JeffPeter** [reply]
Hi,
Thanks for your messages and for the answers! I see that the MOOSE community is really active. 
Jeff

**Reply by crswong888** [reply]
@dschwen The final system of equations solves nodal values of whatever the primary dependent variable is, no? I don't see how it is misleading to say we solve equations at the nodes. And it surely is an interpolation, because cannot know for sure what the values at every point is until we enforce the boundary-value problem and solve the system of equations - at the nodes...

**Reply by hugary1995** [reply]
Hey Chris, you are correct saying "we solve equations at nodes and then interpolate" for Lagrange basis, because Lagrange polynomials satisfy kronecker delta nodewise, so dofs are defined on nodes and the solution happens to have the "interpolation" property.

But as Daniel pointed out, the world isn't just about Lagrange polynomials (even though it is the most commonly used one in FEM). There are other polynomials that do not satisfy those properties. Monomials and Splines are two common examples.

**Reply by crswong888** [reply]
Okay, that's fair. You are right, my perspective of the FEM is quite limited to polynomials that satisfy nodal DOFs - different variations of Hermites at worst.
